openssl-dynamic/src/main/java/io/netty/internal/tcnative/SSLContext.java

```diff
@@ -649,15 +649,33 @@ public static void setAlpnProtos(long ctx, String[] alpnProtos, int selectorFail
     public static native void setUseTasks(long ctx, boolean useTasks);
 
     /**
-     * Set the {@link SSLPrivateKeyMethod} to use for the given {@link SSLContext}. This allows to offload privatekey operations
+     * Set the {@link SSLPrivateKeyMethod} to use for the given {@link SSLContext}.
+     * This allows to offload private key operations
      * if needed.
      *
      * This method is currently only supported when {@code BoringSSL} is used.
      *
      * @param ctx context to use
      * @param method method to use for the given context.
      */
-    public static native void setPrivateKeyMethod(long ctx, SSLPrivateKeyMethod method);
+    public static void setPrivateKeyMethod(long ctx, final SSLPrivateKeyMethod method) {
+        setPrivateKeyMethod(ctx, new AsyncSSLPrivateKeyMethodAdapter(method));
+    }
+
+    /**
+     * Set the {@link AsyncSSLPrivateKeyMethod} to use for the given {@link SSLContext}.
+     * This allows to offload private key operations if needed.
+     *
+     * This method is currently only supported when {@code BoringSSL} is used.
```

Comment:
Should we log a warning once when this method is called without BoringSSL?

Reply:
We already this in netty :)